presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageFileFooterOutput.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.pagefile;
+
+import com.facebook.presto.orc.stream.DataOutput;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.SliceOutput;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+import static java.util.Objects.requireNonNull;
+
+public class PageFileFooterOutput
+        implements DataOutput
+{
+    private final List<Long> stripeOffsets;
+
+    public PageFileFooterOutput(List<Long> stripeOffsets)
+    {
+        this.stripeOffsets = ImmutableList.copyOf(requireNonNull(stripeOffsets, "stripeOffsets is null"));
+        checkArgument(!stripeOffsets.isEmpty(), "stripeOffsets should at least contain offset 0");
+    }
+
+    @Override
+    public long size()
+    {
+        return SIZE_OF_LONG * (stripeOffsets.size());
+    }
+
+    @Override
+    public void writeData(SliceOutput sliceOutput)
+    {
+        for (long offset : stripeOffsets) {
+            sliceOutput.writeLong(offset);
```

Comment:
I see what are you doing. You are treading the footer as a "stripe" adding the last offset as an offset of the footer. It would work, but it is a little confusing. How about we make it more explicit. Let's add offsets only for existing stripes, and then add the size of the footer in bytes as a last `long` of the file. Similar to ORC.

Reply:
I was planning to draw a footer design picture before adding you to review, but you get the idea correctly !  

- Like ORC,  stripOffset is the beginning of a stripe, thus offSet start from 0
- Use N + 1  `long` to represent N stripes has advantage of labeling the end of last stripe.
- The last stripeOffset is offset of footer, it is use for the same purpose as size of footer, but has advantage of avoiding calculation last stripe size. Otherwise, `lastStripeSize = fileSize - beginningOfLastStripe - stripeSize` 
- Maybe we should use size of footer instead of beginning of footer to be consistent with ORC

Reply:
> To me the simplest format would be an offset and length. Why try to do fancy things to derive where things are and how long they are when space is not at a premium?

Length is inferred from difference of adjacent offset, @Andri was suggesting to use footer size to replace footer offset.